src/main/scala/temple/test/ProjectTester.scala

```diff
@@ -17,27 +19,56 @@ object ProjectTester {
 
   /** Configure the infrastructure for testing */
   private def performSetup(templefile: Templefile, generatedPath: String): ProjectConfig = {
-    println("🍪 Spinning up Kubernetes infrastructure...")
-    // Deploy Kubernetes
-    exec(s"sh $generatedPath/deploy.sh")
+    val provider = templefile
+      .lookupMetadata[Metadata.Provider]
+      .getOrElse(throw new RuntimeException("Could not find deployment information"))
```

Comment:
Personally I’d prefer braces here to show that this doesn’t get immediately executed